#9 CRUD for Course Materials

selinuluca opened the pull request:
Implemented CRUD functionality for course materials.
Integrated create, read, update, and delete operations into the system.
Connected CRUD features with the Vue.js and Inertia.js front end.

selinuluca committed:
#9 CRUD for Course Materials

EwelinaSkrzypacz commented on Column('course_materials', 'lesson_id');:
```diff
@@ -0,0 +1,258 @@
+
```
What is this file?

EwelinaSkrzypacz commented on 2024cModelsCourseMaterial::all();:
```diff
@@ -0,0 +1,11 @@
+[90m= [39m[34;4mInterns2024c\Models\CourseMaterial[39;24m {#6442
```
Do we need this file? Why the name of this file look like line of code to get all materials?

EwelinaSkrzypacz commented on 2024cModelsLesson::create([:
```diff
@@ -0,0 +1,12 @@
+[90m= [39m[34;4mIlluminate\Database\Eloquent\Collection[39;24m {#6273
```
The same as above

EwelinaSkrzypacz commented on Sample Lesson,:
Do we need this?

EwelinaSkrzypacz commented on This is a sample lesson.,:
Do we need this?

EwelinaSkrzypacz commented on app/Http/Controllers/CourseMaterialController.php:
```diff
@@ -0,0 +1,149 @@
+<?php
+
```
Missing `declare(strict_types=1);`

EwelinaSkrzypacz replied:
Please add here everywhere, where this line is missing

EwelinaSkrzypacz commented on app/Http/Controllers/CourseMaterialController.php:
```diff
@@ -0,0 +1,149 @@
+
+class CourseMaterialController extends Controller
+{
+    /**
```
Please remove all unnecessary comments

EwelinaSkrzypacz commented on app/Http/Controllers/CourseMaterialController.php:
```diff
@@ -0,0 +1,149 @@
+     */
+    public function index()
+    {
+        $materials = \Interns2024c\Models\CourseMaterial::with('lesson')->get();
```
I believe we can make like this

`$materials = CourseMaterial::with('lesson')->get();`

and please import CourseMaterial model

EwelinaSkrzypacz replied:
And I saw similar thing with `Lesson` model, so please fix that also

EwelinaSkrzypacz commented on app/Http/Controllers/CourseMaterialController.php:
```diff
@@ -0,0 +1,149 @@
+            return $material->lesson->title ?? 'No Lesson';
+        });
+
+        $sortedMaterials = $groupedMaterials->sortBy(function ($group, $lessonTitle) use ($englishLevels) {
```
Missing types - in other places too, so please fix this everywhere

EwelinaSkrzypacz commented on app/Http/Controllers/CourseMaterialController.php:
```diff
@@ -0,0 +1,149 @@
+
+        $sortedMaterials = $groupedMaterials->sortBy(function ($group, $lessonTitle) use ($englishLevels) {
+            $index = array_search($lessonTitle, $englishLevels);
+            return $index !== false ? $index : PHP_INT_MAX;
```
I'm not sure that it should be like that - I mean this `PHP_INT_MAX` thing

EwelinaSkrzypacz commented on app/Http/Controllers/CourseMaterialController.php:
```diff
@@ -0,0 +1,149 @@
+    /**
+     * Display the specified resource.
+     */
+    public function show($id)
```
Missing type and return type - this is not the only place in the code where this occurs. Please review the rest of the code and make the necessary corrections everywhere.

EwelinaSkrzypacz commented on app/Http/Middleware/HandleInertiaRequests.php:
```diff
@@ -31,11 +31,8 @@ public function version(Request $request): ?string
      */
     public function share(Request $request): array
     {
-        return [
```
I think it should stay

EwelinaSkrzypacz commented on app/Models/CourseMaterial.php:
```diff
@@ -0,0 +1,27 @@
+        return $this->belongsTo(\Interns2024c\Models\Lesson::class);
+    }
+}
+
```
Please remove extra line - we should have only one empty line at the end of the file

EwelinaSkrzypacz commented on database/migrations/2024_12_17_215328_create_course_materials_table.php:
```diff
@@ -0,0 +1,33 @@
+    {
+        Schema::dropIfExists('course_materials');
+    }
+};
```
Please add one empty line at the end of the file

EwelinaSkrzypacz commented on resources/js/Pages/CourseMaterials/Create.vue:
```diff
@@ -0,0 +1,91 @@
+<template>
+  <div style="max-width: 600px; margin: 0 auto; padding: 20px; border: 1px solid #ddd; border-radius: 8px; background-color: #f9f9f9;">
+    <h1 style="text-align: center; font-size: 24px; margin-bottom: 20px;">Create New Course Material</h1>
+    <form @submit.prevent="submitForm" style="display: flex; flex-direction: column; gap: 15px;">
```
Please fix all GitHub warnings

EwelinaSkrzypacz commented on resources/js/Pages/CourseMaterials/Create.vue:
```diff
@@ -0,0 +1,91 @@
+</template>
+
+<script>
+import { ref } from 'vue';
```
Please fix all GitHub errors

EwelinaSkrzypacz commented on resources/js/Pages/CourseMaterials/Create.vue:
```diff
@@ -0,0 +1,91 @@
+  </div>
+</template>
+
+<script>
```
Please move <script> to the top - like we have for example in `Register.vue` file

EwelinaSkrzypacz commented on resources/js/Pages/CourseMaterials/Index.vue:
```diff
@@ -0,0 +1,191 @@
+            <button
+                @click="editMaterial(material.id)"
+                style="
+                margin-top: 15px;
```
Please use tailwindcss to style stuff

EwelinaSkrzypacz commented on routes/web.php:
```diff
@@ -16,12 +18,25 @@
     Route::delete("/profile", [ProfileController::class, "destroy"])->name("profile.destroy");
 });
 
+// Debug lessons route
```
Please remove this route - as I understand it's for debugging, so it shouldn't be pushed

EwelinaSkrzypacz commented on storage/app/.gitignore:
Why permissions are changed?

EwelinaSkrzypacz commented on resources/js/Pages/CourseMaterials/Create.vue:
```diff
@@ -0,0 +1,91 @@
+  },
+
+  setup(props) {
+    console.log('Lessons prop (raw):', props.lessons);
```
Please remove this `console.log`

EwelinaSkrzypacz reviewed (commented):
1. Please fix checks - all are failed.
2. Add in navigation button to course materials list - now I had to manually write url to the page
3. I think that we should first merge pr with course and after that we can make change to this code - I mean that for now I can't add course materials because I can't create course now; I see that we have model `Lesson` but I think we should change this to course
4. Add validation in create/edit form - for example, when user go to create course material page and fills nothing, just click "Save"  I see only frontend validation - please add also backend validation and show message
5. Missing empty state; 

![image](https://github.com/user-attachments/assets/9936a8e9-5061-46b2-af49-d1e7f7d1e6cc)

6. Missing title on browser card

![image](https://github.com/user-attachments/assets/e5bebc22-9bf3-4b53-8494-f0dcc97a89b5)

7. Please add screenshots to pr.

8. Please include the task number related to this pull request in the description.

selinuluca committed:
Added LessonSeeder.php

selinuluca committed:
Add navigation button to Course Materials list :
Added a navigation button to the Course Materials list for easier access.
Previously, the URL had to be manually entered to navigate to the page.

EwelinaSkrzypacz reviewed (commented):
I'll be happy to review this PR once all checks have passed and my feedback has been implemented.
